lib/xgu/xgux.h

```diff
@@ -0,0 +1,88 @@
+#ifndef __XGUX_H
+#define __XGUX_H
+
+#include "xgu.h"
+
+inline
+void xgux_draw_arrays(XguPrimitiveType mode, unsigned int start, unsigned int count) {
+    uint32_t *p = pb_begin();
+    p = xgu_begin(p, mode);
+    p = xgu_draw_arrays(p, start, count);
+    p = xgu_end(p);
+    pb_end(p);
+}
+
+inline
+void xgux_draw_elements16(XguPrimitiveType mode, unsigned int count, const uint16_t* elements) {
+    uint32_t *p = pb_begin();
+    p = xgu_begin(p, mode);
+
+    /* Submit elements in pairs if possible */
+    unsigned int pair_count = count / 2;
+    for (unsigned int i = 0; i < pair_count; ) {
+
+        if (i > 0) {
+            /* Start next batch */
+            p = pb_end(p);
+            p = pb_begin();
+        }
```

Comment:
Behaviour not tested on hardware. But we need something like this to avoid `TRIANGLE_STRIP` etc. from breaking (as we can't just restart a new draw call).
If this doesn't work, then this potentially needs more logic, depending on primitive type.

Reply:
This comment is confusing, because this restarts the pb, not the draw call? Needs testing on hardware. I don't remember why we can't just start another draw call on the same pushbuffer (other than potentially running out of memory).